src/ModelManager.ts

```diff
@@ -125,96 +153,170 @@ export class ModelManager {
+     * the library attempts to fetch a fragment of model.
+     *
+     * @fires cq-pagemodel-loaded if root model path is available
+     */
+    public initializeAsync(config?: ModelManagerConfiguration | string): void {
+        this.destroy();
+
+        const modelConfig = this._toModelConfig(config);
+        const initialModel = modelConfig && modelConfig.model;
+
+        this._initializeFields(modelConfig);
+        this._getPathsForModel(modelConfig);
+
+        const { rootModelPath } = this._modelPaths;
+
+        this._modelStore = new ModelStore(rootModelPath, initialModel);
+
+        if (rootModelPath) {
+            this._setInitializationPromise(rootModelPath);
         }
+    }
+
+    /**
+     * Initializes the class fields for ModelManager
+     * If no path is provided, fallbacks are applied in the following order:
+     *
+     */
+    private _initializeFields(config?: ModelManagerConfiguration) {
+        this._listenersMap = {};
+        this._fetchPromises = {};
+        this._initPromise = null;
+
+        this._modelClient = ((config && config.modelClient) || new ModelClient());
 
-        this._clientlibUtil = new AuthoringUtils(this.modelClient.apiHost);
         this._editorClient = new EditorClient(this);
-        this._modelStore = (initialModel) ? new ModelStore(rootModelPath, initialModel) : new ModelStore(rootModelPath);
+        this._clientlibUtil = new AuthoringUtils(this.modelClient.apiHost);
+    }
 
-        this._initPromise = this._checkDependencies().then(() => {
-            const data = this.modelStore.getData(rootModelPath);
+    /**
+     * Returns paths required for fetching root model
+     * Order of preference for determining the root model url -
+     * 1. Page path provided via config
+     * 2. Meta property value for cq:pagemodel_root_url, if set
+     * 3. Model path contained in URL for default SPA.
+     * To add - For remote SPA opened within editor, path contained in parent i.e. AEM url
+     * 4. If none, it defaults to empty string
+     *
+     */
+    private _getPathsForModel(config?: ModelManagerConfiguration) {
+        // Model path explicitly provided by user in config
+        const path = config && config.path;
 
-            if (data && (Object.keys(data).length > 0)) {
-                triggerPageModelLoaded(data);
+        // Model path set statically via meta property
+        const pageModelRoot = PathUtils.getMetaPropertyValue(MetaProperty.PAGE_MODEL_ROOT_URL);
+        const metaPropertyModelUrl = PathUtils.internalize(pageModelRoot);
 
-                return data;
-            } else {
-                return this._fetchData(rootModelURL).then((rootModel: Model) => {
-                    this.modelStore.initialize(rootModelPath, rootModel);
+        const aemApiHost = this.modelClient.apiHost;
+        const isRemoteApp = PathUtils.isBrowser() && aemApiHost && (PathUtils.getCurrentURL() !== aemApiHost);
```

Comment:
wasn't that already defined in `ContentFrame.js`? cc: @CezCz

Reply:
@grubyak @CezCz Sorry, I didn't understand. Could you elaborate please?

Reply:
This is all as it should be from my perspective. Slight code duplication between this and ContentFrame but how to go around it?

Reply:
two criteria in different places to indicate same status, can get out of sync, but i see your point

Reply:
Here the check is independent of whether it's open in editor as well. It's more along the lines of whether we can utilize the pathname, since for remote apps the pathname and the path for model are not the same. Wdyt @grubyak ?

Reply:
> two criteria in different places to indicate same status, can get out of sync, but i see your point

Should we encapsulate that oneliner into a function? and comment

Reply:
Also, the use of `PathUtils.isBrowser()` indicates that it won't work on SSR. Would that be an issue?

Reply:
`PathUtils.getCurrentURL() !== aemApiHost`

I am not certain about that condition. The current URL should contain the host the path and all. The host on the other hand sounds way shorter. If both values contain the same information, we may want to rename fields